code/datums/actions/mobs/create_turrets.dm

```diff
@@ -0,0 +1,114 @@
+/datum/action/cooldown/mob_cooldown/create_turrets
+	name = "Create Sentinels"
+	button_icon = 'icons/mob/simple/lavaland/lavaland_monsters.dmi'
+	button_icon_state = "legion_turret"
+	desc = "Create legion sentinels that fire at any enemies."
+	cooldown_time = 2 SECONDS
+	var/minimum_turrets = 2
+	var/maximum_turrets = 2
+
+/datum/action/cooldown/mob_cooldown/create_turrets/Activate(atom/target_atom)
+	disable_cooldown_actions()
+	create(target_atom)
+	StartCooldown()
+	enable_cooldown_actions()
+	return TRUE
+
+/datum/action/cooldown/mob_cooldown/create_turrets/proc/create(atom/target)
+	playsound(owner, 'sound/magic/RATTLEMEBONES.ogg', 100, TRUE)
+	var/list/possiblelocations = list()
+	for(var/turf/T in oview(owner, 4)) //Only place the turrets on open turfs
+		if(T.is_blocked_turf())
+			continue
+		possiblelocations += T
+	for(var/i in 1 to min(rand(minimum_turrets, maximum_turrets), LAZYLEN(possiblelocations))) //Makes sure aren't spawning in nullspace.
```

Comment:
Does the `1 to min(random number, thing which might be 0)` actually work?
I would have thought that the `LAZYLEN` is only executed the first time, but we're mutating `possiblelocations` so it might become empty during the loop, if there's only one viable turf.

Reply:
also just use `length()`, `possible_locations` isn't a lazylist.